packages/main/src/CalendarLegend.ts

```diff
@@ -0,0 +1,203 @@
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import ItemNavigation from "@ui5/webcomponents-base/dist/delegate/ItemNavigation.js";
+import NavigationMode from "@ui5/webcomponents-base/dist/types/NavigationMode.js";
+import {
+	isDown,
+	isUp,
+} from "@ui5/webcomponents-base/dist/Keys.js";
+import ItemNavigationBehavior from "@ui5/webcomponents-base/dist/types/ItemNavigationBehavior.js";
+import CalendarLegendItemType from "./types/CalendarLegendItemType.js";
+import CalendarLegendTemplate from "./generated/templates/CalendarLegendTemplate.lit.js";
+import CalendarLegendItem from "./CalendarLegendItem.js";
+
+// Styles
+import CalendarLegendCss from "./generated/themes/CalendarLegend.css.js";
+
+type CalendarLegendItemSelectionChangeEventDetail = {
+	item: CalendarLegendItem;
+}
+
+/**
+ * @class CalendarLegend
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * The <code>ui5-calendar-legend</code> component is designed for use within the <code>ui5-calendar</code> to display a legend.
+ * Each <code>ui5-calendar-legend-item</code> represents a unique date type, specifying its visual style
+ * and a corresponding textual label.
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import "@ui5/webcomponents/dist/CalendarLegend.js";</code>
+ *
+ * @slot items - Defines the items of the component.
+ *
+ * @constructor
+ * @extends UI5Element
+ * @public
+ * @since 1.22.0
+ */
+@customElement({
+	tag: "ui5-calendar-legend",
+	renderer: litRender,
+	styles: CalendarLegendCss,
+	template: CalendarLegendTemplate,
+	dependencies: [CalendarLegendItem],
+})
+@event("_calendar-legend-selection-change", {
+	detail: {
+		item: { type: CalendarLegendItem },
+	},
+})
+class CalendarLegend extends UI5Element {
+	/**
+	 * Hides the Today item in the legend.
+	 *
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideToday!: boolean;
+
+	/**
+	 * Hides the Selected day item in the legend.
+	 *
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideSelectedDay!: boolean;
+
+	/**
+	 * Hides the Non-Working day item in the legend.
+	 *
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideNonWorkingDay!: boolean;
+
+	/**
+	 * Hides the Working day item in the legend.
+	 *
+	 * @defaultvalue false
+	 * @public
+	 */
+	@property({ type: Boolean })
+	hideWorkingDay!: boolean;
+
+	/**
+	 * Defines the items of the component.
+	 * @slot items
+	 * @public
+	 */
+	@slot({
+		type: HTMLElement,
+		invalidateOnChildChange: true,
+		individualSlots: true,
+		"default": true,
+	 })
+	items!: Array<CalendarLegendItem>;
+
+	_itemNavigation: ItemNavigation;
+
+	constructor() {
+		super();
+
+		this._itemNavigation = new ItemNavigation(this, {
+			navigationMode: NavigationMode.Horizontal,
+			behavior: ItemNavigationBehavior.Static,
+			getItemsCallback: () => this.focusableElements,
+		});
+	}
+
+	_onMouseDown(e: MouseEvent) {
```

Comment:
I would add private click event which is fired by calendar legend item when it's clicked. Mouse down doesn't mean that something is clicked so this rules is not correct to move the focus to the current item.

At all moving the listeners directly to the root and checking the target looks better.

Reply:
As discussed offline, we'll leave it with mousedown as it is in UI5.